src/controllers/resource-store/index.tsx

```diff
@@ -89,101 +135,205 @@ export const actions: Actions = {
       cached.accessedAt = getAccessedAt();
       dispatch(setResourceState(type, key, cached));
 
-      return cached;
+      return Promise.resolve(cached);
     }
 
     return dispatch(
-      getResourceFromRemote(resource, routerStoreContext, options)
+      privateActions.getResourceFromRemote(
+        resource,
+        routerStoreContext,
+        options
+      )
```

Comment:
On reflection I think we should revert changes to this function because of how it effects getting all resources on the StaticRouter.

When we have the `async` keyword in place then errors thrown by `getKey()` will appear as a rejecting promise thanks to the `Promise.all()` for getting all resources. However with this chance errors thrown by `getKey()` will just appear as a hard error.

So leaving this change in place would mean that getting all resources would need to cater for both rejecting promise and hard errors which is a bit unfortunate.